File: LeetCode-Problems/Easy/Shuffle-Array.java
//Java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int[] firstHalf = new int[n];
        int[] secondHalf = new int[n];
        int[] returnedArray = new int[nums.length];
        for (int i = 0; i < n; i++){
            firstHalf[i] = nums[i];
        }
        int x = 0;
        for (int i = n; i < nums.length; i++){
            secondHalf[x] = nums[i];
            x++;
        }
        int j = 0;
        for (int i = 0; i < nums.length; i+=2){
            returnedArray[i] = firstHalf[j];
            returnedArray[i+1] = secondHalf[j];
            j++;
        }
        return returnedArray;
        
    }
}

File: LeetCode-Problems/Easy/Shuffle-Array.py
# Python
class Solution(object):
    def shuffle(self, nums, n):
        """
        :type nums: List[int]
        :type n: int
        :rtype: List[int]
        """
        firstHalf = list()
        secondHalf = list()
        returnedList = list()
        for x in range(0, len(nums)/2):
            firstHalf.append(nums[x])
        for x in range(len(nums)/2, len(nums)):
            secondHalf.append(nums[x])
        
        i = 0;
        for x in range(0, len(nums)/2):
            returnedList.append(firstHalf[i])
            returnedList.append(secondHalf[i])
            i += 1
        return returnedList
